Global styles: do not navigate twice to home screen when opening the sidebar

ciampo opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Extracted from #64777

Prevent `Navigator` from navigating twice to the home screen when opening the Global Styles sidebar.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

The double navigation cause the first screen displayed in the Global Styles sidebar to animate, which is not the expected behavior (the enter animation should be skipped on the first screen)

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

By inverting the logic of a check inside a `useEffect` in the global styles sidebar code.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

- (important) Reload the site editor
- Open the global styles sidebar
- Make sure that the initial navigator screen doesn't slide from the right when opening the sidebar

## Screenshots or screencast <!-- if applicable -->

| Before (`trunk`) | After (this PR) |
|---|---|
| <video src="https://github.com/user-attachments/assets/d53d07c7-4cf2-4454-bfa8-cb0c501e85ed" /> | <video src="https://github.com/user-attachments/assets/8a227c69-84c0-425a-8953-b9a9842e2404" /> |

ciampo committed:
Global styles: do not navigate twice to home screen on sidebar load

ciampo commented:
@ramonjd @andrewserong @Mamaduka I'm not really sure if this change introduces unintended regressions, but I figured you are the most indicated reviewers as you worked on https://github.com/WordPress/gutenberg/pull/59261

ramonjd commented:
Hmm I'm seeing a max depth exceeded in the `InspectorControlsColorSlot` when selecting a block and opening its settings in the rhs

https://github.com/user-attachments/assets/eb46f036-996d-4ca3-a074-91328d9c57fa

I'll see if I can get more info...

andrewserong commented:
> Hmm I'm seeing a max depth exceeded in the InspectorControlsColorSlot when selecting a block and opening its settings in the rhs

I was seeing that, too. I've just rebased against `trunk` and the error has disappeared... I wonder if there was a different bug somewhere that was causing this?

ramonjd commented on packages/edit-site/src/components/global-styles/ui.js:
```diff
@@ -280,10 +280,9 @@ function GlobalStylesEditorCanvasContainerLink() {
-				if ( path !== '/' && ! isRevisionsOpen ) {
-					return;
+				if ( path !== '/' && isRevisionsOpen ) {
+					goTo( '/' );
```
Hmm, in my testing so far, I think we could get right of the entire default block.


https://github.com/user-attachments/assets/2f49412f-cc51-4d16-aa36-26c702d458fe

ciampo replied:
Me likes more red! Pushed an update with the whole `default` block removed

ramonjd commented:
> I've just rebased against trunk and the error has disappeared... I wonder if there was a different bug somewhere that was causing this?


Good call! Thanks

ciampo committed:
Remove whole default block

ciampo commented:
Rebased on top of latest `trunk` (which should include [the fix](https://github.com/WordPress/gutenberg/pull/65533) mentioned above), and removed the whole `default` block as suggested by @ramonjd .

I guess we can give it one last round of checks and hopefully merge if everything keeps looking good 🤞

andrewserong reviewed (approved):
Thanks for the update, all working nicely for me now! 🚀

ramonjd commented:
LGTM! I'll merge
